src/clientiface.cpp

```diff
@@ -46,396 +45,245 @@ const char *ClientInterface::statenames[] = {
 	"SudoMode",
 };
 
-
-
 std::string ClientInterface::state2Name(ClientState state)
 {
 	return statenames[state];
 }
 
-RemoteClient::RemoteClient() :
-	m_max_simul_sends(g_settings->getU16("max_simultaneous_block_sends_per_client")),
-	m_min_time_from_building(
-		g_settings->getFloat("full_block_send_enable_min_time_from_building")),
-	m_max_send_distance(g_settings->getS16("max_block_send_distance")),
-	m_block_optimize_distance(g_settings->getS16("block_send_optimize_distance")),
-	m_max_gen_distance(g_settings->getS16("max_block_generate_distance")),
-	m_occ_cull(g_settings->getBool("server_side_occlusion_culling"))
-{
-}
-
-void RemoteClient::ResendBlockIfOnWire(v3s16 p)
-{
-	// if this block is on wire, mark it for sending again as soon as possible
-	if (m_blocks_sending.find(p) != m_blocks_sending.end()) {
-		SetBlockNotSent(p);
-	}
-}
-
-LuaEntitySAO *getAttachedObject(PlayerSAO *sao, ServerEnvironment *env)
-{
-	if (!sao->isAttached())
-		return nullptr;
-
-	int id;
-	std::string bone;
-	v3f dummy;
-	bool force_visible;
-	sao->getAttachment(&id, &bone, &dummy, &dummy, &force_visible);
-	ServerActiveObject *ao = env->getActiveObject(id);
-	while (id && ao) {
-		ao->getAttachment(&id, &bone, &dummy, &dummy, &force_visible);
-		if (id)
-			ao = env->getActiveObject(id);
-	}
-	return dynamic_cast<LuaEntitySAO *>(ao);
-}
-
-void RemoteClient::GetNextBlocks (
-		ServerEnvironment *env,
-		EmergeManager * emerge,
-		float dtime,
-		std::vector<PrioritySortedBlockTransfer> &dest)
+/*WMSSuggestion RemoteClient::getNextBlock(
+		EmergeManager *emerge, ServerFarMap *far_map)*/
+WMSSuggestion RemoteClient::getNextBlock(EmergeManager *emerge)
 {
-	// Increment timers
-	m_nothing_to_send_pause_timer -= dtime;
-
-	if (m_nothing_to_send_pause_timer >= 0)
-		return;
-
-	RemotePlayer *player = env->getPlayer(peer_id);
-	// This can happen sometimes; clients and players are not in perfect sync.
-	if (!player)
-		return;
+	// If the client has not indicated it supports the new algorithm, fill in
+	// autosend parameters and things should work fine
+	if (m_fallback_autosend_active)
+	{
+		s16 radius_map = g_settings->getS16("max_block_send_distance");
+		s16 radius_far = 0; // Old client does not understand FarBlocks
+		float far_weight = 3.0f; // Whatever non-zero
+		float fov = 1.72f; // Assume something (radians)
+		u32 max_total_mapblocks = 5000; // 5000 is the default client-side setting
+		u32 max_total_farblocks = 0;
+
+		// One version of Minetest supports wanted_range and camera_fov via
+		// PlayerSAO. Try to use them.
```

Comment:
(all 5.x versions support this)
```suggestion
		// If the SAO exists, get wanted_range and camera_fov from there.
```